sdk/python/feast/pyspark/historical_feature_retrieval_job.py

```diff
@@ -412,10 +420,12 @@ def start_job(spark: SparkSession, conf: Dict):
 
 
 if __name__ == "__main__":
-    spark = SparkSession.builder.appName("Batch Retrieval").getOrCreate()
-    spark.sparkContext.addFile("config.json")
-    config_file_path = SparkFiles.get("config.json")
-    with open(config_file_path, "r") as config_file:
-        conf = json.load(config_file)
-        start_job(spark, conf)
+    spark = SparkSession.builder.getOrCreate()
+    parser = argparse.ArgumentParser(description="Retrieval job arguments")
+    parser.add_argument(
+        "config_json", type=str, help="Configuration in json string format"
```

Comment:
So this is a json blob right?

Reply:
Yes, it will be a json string, similar approach to the Spark jobs written by @pyalex . Originally i chose to use an actual config file, but it makes the job launcher much more complicated due to uploading and retrieval of configuration file.

Reply:
In case of `ingestion job` there're several json argument and I suggest you can split your config as well. It's just easier to work with job that has separated parameters, that one big json.
So in `ingestion-job` there are
`--feature-table` - json with project, name, entities, features
`--source` separate source to be able to run custom source
I assume in historical retrieval you would also need
`--entity-source`

Here's example of arguments for `ingestion job`
```

--source "{"kafka":{"bootstrapServers":"10.200.219.238:6668","topic":"supply-aggregator-sauron","classpath":"com.gojek.esb.aggregate.supply.AggregatedSupplyMessage","mapping":{},"timestampColumn":"event_timestamp"}}" 
--feature-table "{"name":"file","project":"default","entities":[{"name":"customer","type":"STRING"}],"features":[{"name":"feature1","type":"INT32"}]}" 
```

Reply:
I don't think that our jobs arguments will be completely the same. Historical job receives several `feature tables` for example. But at least we can try to keep it similar

Reply:
`--source separate source to be able to run custom source`

@pyalex How would the custom source applies in the context of historical feature retrieval? Is it something like a user input to override the Feature Table source registered on Feast Core?

Reply:
Or do you mean that the source of the feature tables should be it's own parameter (`--source`), rather than as a field under `--feature-table` ?

Reply:
Just to throw in another idea, for feature_table and entity we could also pass the protobuf in JSON format. You'd still to have to manually parse it, but on the caller side it would be easier as we could replace [custom serialization code](https://github.com/feast-dev/feast/pull/1026/files#diff-7bc0d6431261a5482c3078341f05acf6R134-R156) with e.g. `json.dumps(MessageToDict(feature_table.to_proto()))`

Reply:
@khorshuheng I just provided example of ingestion job args. It may be not very relevant to Historical directly. I'm just saying we can try to converge arguments format.

Reply:
@oavdeev we can move towards protobuf-like format. But it's already can't be 100% compatible. FeatureTable, for example, has entities as strings. Whereas most jobs need type as well. So some massaging would still be required

Reply:
I see, didn't realize that. Let's just go with bespoke json then like in the ingestion job